worlds/lingo/locations.py

```diff
@@ -0,0 +1,61 @@
+from typing import Dict, NamedTuple, Optional, List
+from BaseClasses import Location
+from .static_logic import StaticLingoLogic, RoomAndPanel, Room
+
+
+class LocationData(NamedTuple):
+    """
+    LocationData for a location in Lingo
+    """
+    code: Optional[int]
+    room: str
+    panels: List[RoomAndPanel]
+    include_reduce: bool
+
+    def panel_ids(self):
+        ids = set()
+        for panel in self.panels:
+            effective_room = self.room if panel.room is None else panel.room
+            panel_data = StaticLingoLogic.PANELS_BY_ROOM[effective_room][panel.panel]
+            ids = ids | set(panel_data.internal_ids)
+        return ids
+
+
+class LingoLocation(Location):
+    """
+    Location from the game Lingo
+    """
+    game: str = "Lingo"
+
+
+class StaticLingoLocations:
+    """
+    Defines the locations that can be included in a Lingo world
+    """
+
+    base_id: int = 0
```

Comment:
Same comments here as Items.py. I'm really concerned about your `ALL_ITEMS_TABLE` and `ALL_LOCATION_TABLE` with multiple lingo worlds but I assume you've actually tested that? It just looks really unsafe.